golog_test.go

```diff
@@ -115,6 +117,21 @@ func TestDebugJson(t *testing.T) {
 	}
 }
 
+func TestDebugZap(t *testing.T) {
+	observedZapCore, observedLogs := observer.New(zap.DebugLevel)
+	zl := zap.New(observedZapCore)
+
+	SetOutput(ZapOutput(zl))
+	l := LoggerFor("myprefix")
+	l.Debug("Hello world")
+	defer ops.Begin("name").Set("cvarA", "a").Set("cvarB", "b").End()
+	l.Debugf("Hello %v", true)
+	entries := observedLogs.All()
+	assert.Equal(t, 2, len(entries))
+	assert.Equal(t, "Hello world", entries[0].Message)
+	assert.Equal(t, "name", entries[1].ContextMap()["op"])
```

Comment:
Maybe also add tests for Error logging and also check that line numbers look correct?